validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -38,54 +53,323 @@ type SetupConfig struct {
+			return nil, fmt.Errorf("no file exists in remote signer key file path %s", km.keyFilePath)
+		}
+	}
+
+	var ppk []string
+	// load key values
+	if cfg.PublicKeysURL != "" {
+		providedPublicKeys, err := km.client.GetPublicKeys(ctx, cfg.PublicKeysURL)
 		if err != nil {
 			erroredResponsesTotal.Inc()
-			return nil, errors.Wrap(err, fmt.Sprintf("could not get public keys from remote server url: %v", km.publicKeysURL))
+			return nil, errors.Wrapf(err, "could not get public keys from remote server url: %v", cfg.PublicKeysURL)
+		}
+		ppk = providedPublicKeys
+	} else if len(cfg.ProvidedPublicKeys) != 0 {
+		ppk = cfg.ProvidedPublicKeys
+	}
+
+	// use a map to remove duplicates
+	flagLoadedKeys := make(map[string][48]byte)
+
+	// Populate the map with existing keys
+	for _, key := range ppk {
+		decodedKey, err := hexutil.Decode(key)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not decode public key for remote signer")
+		}
+		if len(decodedKey) != fieldparams.BLSPubkeyLength {
+			return nil, fmt.Errorf("invalid public key length: expected %d, got %d", fieldparams.BLSPubkeyLength, len(decodedKey))
+		}
+		flagLoadedKeys[key] = bytesutil.ToBytes48(decodedKey)
+	}
+	km.flagLoadedKeysMap = flagLoadedKeys
+
+	// load from file
+	if keyFileExists {
+		log.WithField("file", km.keyFilePath).Info("Loading keys from file")
+		_, fileKeys, err := km.readKeyFile()
+		if err != nil {
+			return nil, errors.Wrap(err, "could not read key file")
+		}
+		if len(flagLoadedKeys) != 0 {
+			log.WithField("no_flag_loaded_keys", len(flagLoadedKeys)).Info("Combining flag loaded keys and file loaded keys.")
+			maps.Copy(fileKeys, flagLoadedKeys)
+			if err = km.savePublicKeysToFile(fileKeys); err != nil {
+				return nil, errors.Wrap(err, "could not save public keys to file")
+			}
+		}
+		km.lock.Lock()
+		km.providedPublicKeys = maps.Values(fileKeys)
+		km.lock.Unlock()
+		// create a file watcher
+		go func() {
+			err = km.refreshRemoteKeysFromFileChangesWithRetry(ctx, retryDelay)
+			if err != nil {
+				log.WithError(err).Error("could not refresh remote keys from file changes")
+			}
+		}()
+	} else {
+		km.lock.Lock()
+		km.providedPublicKeys = maps.Values(flagLoadedKeys)
+		km.lock.Unlock()
+	}
+
+	return km, nil
+}
+
+func (km *Keymanager) refreshRemoteKeysFromFileChangesWithRetry(ctx context.Context, retryDelay time.Duration) error {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+	if km.retriesRemaining == 0 {
+		return errors.New("file check retries remaining exceeded")
+	}
+	err := km.refreshRemoteKeysFromFileChanges(ctx)
+	if err != nil {
+		km.updatePublicKeys(maps.Values(km.flagLoadedKeysMap)) // update the keys to flag provided defaults
+		km.retriesRemaining--
+		log.WithError(err).Debug("could not refresh remote keys from file changes")
+		log.WithFields(logrus.Fields{"path": km.keyFilePath, "retries_remaining": km.retriesRemaining, "retry_delay": retryDelay}).Warnf("Key file does not exist. Please create a new one. Retrying ...")
```

Comment:
```suggestion
		log.WithFields(logrus.Fields{"path": km.keyFilePath, "retriesRemaining": km.retriesRemaining, "retryDelay": retryDelay}).Warnf("Key file does not exist. Please create a new one. Retrying ...")
```
- log fields are camel cased
- how do you know the error is caused precisely because the file does not exist?